packages/integrations/sitemap/src/utils/parse-i18n-url.ts

```diff
@@ -0,0 +1,42 @@
+interface ParsedI18nUrl {
+	locale: string;
+	path: string;
+}
+
+// NOTE: The parameters have been schema-validated with Zod
+export function parseI18nUrl(
```

Comment:
This utility is pretty much the same as the `parseUrl` utility (deleted in `parse.url.ts`). But I refactored the code to work slightly faster.

Reply:
I bet one of the bottlenecks was the double `.some` over `localCodes` 🤔

Reply:
Yeah that was also very likely too. The first can be skipped because we already validated with Zod.